static/js/disease/chart.ts

```diff
@@ -17,14 +17,253 @@ import * as d3 from "d3";
 import _ from "lodash";
 const SVGNS = "http://www.w3.org/2000/svg";
 const XLINKNS = "http://www.w3.org/1999/xlink";
-export interface DiseaseGeneAssociationData {
+const MARGIN = { top: 30, right: 30, bottom: 90, left: 160 };
+// graph specific dimensions - not all of them are used in the current graph, but might be used in future graphs
+const GRAPH_HEIGHT_XS = 130;
+const GRAPH_HEIGHT_S = 200;
+const GRAPH_HEIGHT_M = 400;
+const GRAPH_WIDTH_S = 660;
+const GRAPH_WIDTH_M = 700;
+const GRAPH_WIDTH_L = 760;
+const GRAPH_WIDTH_XL = 1050;
+// number to select top data points for large data
+const NUM_DATA_POINTS = 10;
+// number by which y axis domain is increased/decreased for y scale to fit all error bars well
+const Y_AXIS_LIMIT = 0.5;
+// dimension of tissue legend dots
+const LEGEND_CIRCLE_RADIUS = 4;
+// tooltip constant for all charts
+const TOOL_TIP = d3.select("#main").append("div").attr("class", "tooltip");
+// horizontal barcharts - default brightness
+const DEFAULT_BRIGHTEN_PERCENTAGE = "112%";
+type Datum = DiseaseGeneAssociationIntervalData;
+/**
+ * When mouse first enters element specified by given id, brighten it and update/display the global tooltip.
+ */
+function onMouseOver(
+  elementID: string,
+  toolTipText: string,
+  brightenPercentage: string = DEFAULT_BRIGHTEN_PERCENTAGE
+): void {
+  // brighten element: https://stackoverflow.com/a/69610045
+  d3.select(`#${elementID}`).style(
+    "filter",
+    `brightness(${brightenPercentage})`
+  );
+  // update tooltip text
+  TOOL_TIP.html(toolTipText);
+  // show tooltip
+  TOOL_TIP.style("opacity", 1);
+}
+
+/**
+ * Update position of global tooltip to track mouse.
+ */
+function onMouseMove(): void {
+  TOOL_TIP.style("left", d3.event.pageX - 60 + "px").style(
+    "top",
+    d3.event.pageY - 60 + "px"
+  );
+}
+
+/**
+ * When mouse leaves element specified by given id, reset its brightness and hide the global tooltip.
+ */
+function onMouseOut(elementID: string): void {
+  // reset element brightness
+  d3.select(`#${elementID}`).style("filter", "brightness(100%)");
+  // hide tooltip
+  TOOL_TIP.style("opacity", 0);
+}
+
+/**
+ * On mouse hover, select hovered element and
+ *  1) highlight it
+ *  2) update the global tooltip
+ *  3) show the global tooltip.
+ *
+ * Unhighlight and hide the global tooltip when the mouse leaves.
+ */
+function handleMouseEvents(
```

Comment:
maybe for all the shared functions and their helper functions, we can extract them out into a bio_charts_utils.ts file and put that file in the shared folder? What do you think @beets 

Some other ideas:
- take the chart.ts from protein folder and the chart.ts from disease folder, rename them to protein_charts.ts and disease_charts.ts, put them in the chart folder and add another bio_charts_utils.ts file
- Make a bio folder that holds both the disease and protein folders and add a shared (or utils) folder to hold bio_chart_utils.ts

Reply:
a bio folder sounds good to me (similar to tools/). i would add that it would be great to have shared functionality between both these pages shared too - please share code between components when you can, especially since these are so similar. these can be done in a follow up PR to keep the refactoring separate - please do it soon.